### PR TITLE
feat: new unstake algorithm

### PR DESCRIPTION
**Pull request by 0xdefacto**

### CODE REVIEW

File: contracts/linear/src/validator_pool.rs [hunk 1]
```diff
@@ -220,6 +230,79 @@ impl ValidatorPool {
         (candidate, amount_to_stake)
     }
 
+    pub fn get_candidate_to_unstake_v1(
```
Comment (dongcool) [inline contracts/linear/src/validator_pool.rs:233]: v1 is ambiguous here, v2 might be better

Comment (dongcool) [reply contracts/linear/src/validator_pool.rs:233]: let's add a brief explanation of the new algorithm in comments as well

File: contracts/linear/src/validator_pool.rs [hunk 2]
```diff
@@ -636,6 +736,13 @@ impl VValidator {
             VValidator::Current(v) => v,
         }
     }
+
+    pub fn get_weight(&self) -> u16 {
+        match self {
+            VValidator::V0(v) => v.weight,
```
Comment (dongcool) [inline contracts/linear/src/validator_pool.rs:742]: what about rename `VValidator` -> `VersionedValidator`?

Comment (0xdefacto) [reply contracts/linear/src/validator_pool.rs:742]: That's clearer definitely. But it's independent of this PR. I make another PR to fix this!
https://github.com/linear-protocol/LiNEAR/pull/141

**Pull request closed without merging**